crates/neon/src/types_impl/promise.rs

```diff
@@ -49,9 +49,74 @@ const BOUNDARY: FailureBoundary = FailureBoundary {
     feature = "promise-api",
     deprecated = "`promise-api` feature has no impact and may be removed"
 )]
-/// The JavaScript [`Promise`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise) value.
+/// The type of JavaScript
+/// [`Promise`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise)
+/// objects.
 ///
-/// [`JsPromise`] may be constructed with [`Context::promise`].
+/// [`JsPromise`] instances may be constructed with [`Context::promise`].
+///
+/// # Example
+///
+/// ```
+/// # use neon::prelude::*;
+/// fn resolve_promise(mut cx: FunctionContext) -> JsResult<JsPromise> {
+///     let (deferred, promise) = cx.promise();
+///     let msg = cx.string("Hello, World!");
+///
+///     deferred.resolve(&mut cx, msg);
+///
+///     Ok(promise)
+/// }
+/// ```
+///
+/// # Example: Asynchronous task
+///
+/// This example uses the [linkify](https://crates.io/crates/linkify) crate in an
+/// asynchronous task, i.e. a
+/// [Node worker pool](https://nodejs.org/en/docs/guides/dont-block-the-event-loop/)
+/// thread, to find all the links in a text string.
+///
+/// More robust implementations might use a custom Rust thread or thread pool to avoid
+/// blocking the worker pool; for more information, see the [`JsFuture`] example.
+///
+/// ```
+/// # use neon::prelude::*;
+/// use linkify::{LinkFinder, LinkKind};
+/// use easy_cast::Cast; // for safe numerical conversions
+///
+/// fn linkify(mut cx: FunctionContext) -> JsResult<JsPromise> {
+///     let text = cx.argument::<JsString>(0)?.value(&mut cx);
+///
+///     let promise = cx
+///         .task(move || {
+///             let (indices, kinds): (Vec<_>, Vec<_>) = LinkFinder::new()
+///                 .spans(&text)
+///                 .map(|span| {
+///                     let end: u32 = span.end().cast();
```

Comment:
Don't you need both the `start` and `end` points? Consider the following text:

> Learn more at https://neon-bindings.com!

This well let you know that it ends at `39`, but there's not enough context to know that the link starts at `14`.

Reply:
The `LinkFinder::spans()` method fully partitions the text into contiguous chunks, including both links and non-links. So it's always guaranteed to start with `0`.

Reply:
That's a little nonobvious so I'll add a clarifying comment.